flang/tools/bbc/bbc.cpp

```diff
@@ -218,24 +219,29 @@ static void convertFortranSourceToMLIR(
     llvm::errs() << "could not open output file " << outputName << '\n';
     return;
   }
-  if (emitFIR) {
-    // Do lowering, but nothing else. Dump FIR and exit.
-    printModule(mlirModule, out);
-    return;
-  }
 
   // Otherwise run the default passes.
   mlir::PassManager pm(mlirModule.getContext());
   mlir::applyPassManagerCLOptions(pm);
   if (passPipeline.hasAnyOccurrences()) {
     passPipeline.addToPipeline(pm);
+  } else if (emitFIR) {
+    // --emit-fir: Build the IR, verify it, and dump the IR (unconditionally).
+    pm.addPass(std::make_unique<Fortran::lower::VerifierPass>());
+    if (mlir::failed(pm.run(mlirModule)))
+      llvm::errs() << "FATAL: verification of lowering to FIR failed";
```

Comment:
Does this change inhibit `-emit-fir` output if the FIR is invalid?  If yes, that might make it harder to analyze problems.

Reply:
This should print the MLIR output regardless of success or failure. I did a little spot test on 1 build config.